File: ios/ReservationApp/ReservationApp/Presentation/Views/Admin/Menu/Components/MenuCategoryTabsView.swift
//
//  MenuCategoryTabsView.swift
//  ReservationApp
//
//  Created by Mehmet Ali Kısacık on 13.11.2025.
//

import SwiftUI

struct MenuCategoryTabsView: View {
    @Binding var selectedTab: CategoryTab
    let onTabSelected: (CategoryTab) -> Void
    
    var body: some View {
        ScrollView(.horizontal, showsIndicators: false) {
            HStack(spacing: ThemeManager.shared.spacing.sm) {
                ForEach(CategoryTab.allCases, id: \.self) { tab in
                    MenuCategoryTabButton(
                        title: tab.displayName,
                        isSelected: selectedTab == tab,
                        action: {
                            selectedTab = tab
                            onTabSelected(tab)
                        }
                    )
                }
            }
            .padding(.horizontal, ThemeManager.shared.spacing.md)
        }
    }
}

struct MenuCategoryTabButton: View {
    let title: String
    let isSelected: Bool
    let action: () -> Void
    
    var body: some View {
        Button(action: action) {
            Text(title)
                .font(AppTypography.body1())
                .fontWeight(isSelected ? .semibold : .regular)
                .foregroundColor(isSelected ? AppColors.primaryMain : AppColors.textSecondary)
                .padding(.horizontal, ThemeManager.shared.spacing.lg)
                .padding(.vertical, ThemeManager.shared.spacing.md)
                .background(isSelected ? AppColors.backgroundActiveTab : Color.clear)
                .cornerRadius(ThemeManager.shared.borderRadius.button.value)
                .overlay(
                    RoundedRectangle(cornerRadius: ThemeManager.shared.borderRadius.button.value)
                        .stroke(isSelected ? Color.clear : AppColors.borderDefault, lineWidth: 1)
                )
        }
        .buttonStyle(PlainButtonStyle())
    }
}

File: android/app/src/main/java/com/reservationapp/ui/admin/menu/MenuCategoryTabsView.kt
package com.reservationapp.ui.admin.menu

import androidx.compose.foundation.background
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.reservationapp.core.ui.theme.*
import com.reservationapp.ui.admin.CategoryTab

@Composable
fun MenuCategoryTabsView(
    selectedTab: CategoryTab,
    onTabSelected: (CategoryTab) -> Unit,
    modifier: Modifier = Modifier
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .horizontalScroll(rememberScrollState())
            .padding(horizontal = Spacing.md),
        horizontalArrangement = Arrangement.spacedBy(Spacing.sm)
    ) {
        CategoryTab.entries.forEach { tab ->
            MenuCategoryTabButton(
                title = tab.displayName,
                isSelected = selectedTab == tab,
                onClick = {
                    onTabSelected(tab)
                }
            )
        }
    }
}

@Composable
private fun MenuCategoryTabButton(
    title: String,
    isSelected: Boolean,
    onClick: () -> Unit
) {
    if (isSelected) {
        Button(
            onClick = onClick,
            modifier = Modifier.clip(RoundedCornerShape(8.dp)),
            colors = ButtonDefaults.buttonColors(
                containerColor = BackgroundActiveTab,
                contentColor = PrimaryMain
            ),
            contentPadding = PaddingValues(horizontal = Spacing.lg, vertical = Spacing.md)
        ) {
            Text(
                text = title,
                style = MaterialTheme.typography.bodyLarge.copy(
                    fontWeight = FontWeight.SemiBold
                )
            )
        }
    } else {
        OutlinedButton(
            onClick = onClick,
            modifier = Modifier.clip(RoundedCornerShape(8.dp)),
            colors = ButtonDefaults.outlinedButtonColors(
                contentColor = TextSecondary
            ),
            border = androidx.compose.foundation.BorderStroke(1.dp, BorderDefault),
            contentPadding = PaddingValues(horizontal = Spacing.lg, vertical = Spacing.md)
        ) {
            Text(
                text = title,
                style = MaterialTheme.typography.bodyLarge.copy(
                    fontWeight = FontWeight.Normal
                )
            )
        }
    }
}

